packages/@aws-cdk/aws-elasticloadbalancing/lib/load-balancer.ts

```diff
@@ -141,10 +141,20 @@ export interface HealthCheck {
   readonly timeout?: Duration;
 }
 
+/**
+ * An object that has instance object.
+ */
+interface IInstance {
+  /**
+   * Ec2 instance
+   */
+  readonly ec2Instance?: Instance
+}
+
 /**
  * Interface that is going to be implemented by constructs that you can load balance to
  */
-export interface ILoadBalancerTarget extends IConnectable {
+export interface ILoadBalancerTarget extends IConnectable, IInstance {
```

Comment:
Looks like what you did is the following:

* Every `ILoadBalancerTaraget` maybe has an instance associated with it. 
* If it does, the `LoadBalancer` will pick it up and add it to the `instanceIds` array.

It's a nice and pragmatic solution, and one that we follow in a lot of places in the CDK already!

Having said that, I'm not suuuper fond of it, mostly for theoretical and aesthetic reasons that have no bearing on the practicality of it. But bear with me for a bit 😆.

---------

Ultimately, what we are trying to achieve is:

* Some load balancer targets may need to add to the `instanceIds` array on `LoadBalancer`.

As far as I can tell, we can do this in one of three ways:

* `ILoadBalancerTarget`s have a public member that the `LoadBalancer` knows to query for, and do something with (the solution you came up with)
* The `attachToClassicLb()` function *returns* a value that the `LoadBalancer` knows what to do with (this is very similar to the first solution, but I like it a bit more because it's more explicitly part of the "attach to Load Balancer" protocol, and less of what it means to *be* a load balancer target -- I told you my objections were going to be very conceptual).
* The load balancer target can *tell* the `LoadBalancer` to add a value to its `instanceIds` array, via some command.

From those three solutions, I like the 3rd one the best. The reason for that is that is that [Tell, Don't Ask](https://martinfowler.com/bliki/TellDontAsk.html) tends to produce software that is more flexible and extensible. 

For example, if we had:

```ts
class LoadBalancer {
  /**
   * @internal
   */
  public _addInstanceId(instanceId: string) { this.instanceIds.push(instanceId); }
}

class InstanceTarrget {
  public attachToClassicLb(lb) {
    lb._addInstanceId(this.instance.instanceId);
  }
}
```

And we said "hey you know what I want to do? I want to have a single target that represents 3 instances", we could easily make that as follows:

```ts
class ThreeInstancesTarget {
  public attachToClassicLb(lb) {
    lb._addInstanceId(this.instance1.instanceId);
    lb._addInstanceId(this.instance2.instanceId);
    lb._addInstanceId(this.instance3.instanceId);
  }
}
```

Whereas if we had the following protocol:

```ts
interface ILoadBalancerTarget {
  public instance?: IInstance;
}
```

Then now there is no way anymore to implement the class `ThreeInstancesTarget`: every target can add exactly 0 or 1 instances, no more.

Of course, we could predict this use case and type it as follows:

```ts
interface ILoadBalancerTarget {
  public instance?: IInstance[];
}
```

And now every load balancer target can have 0, 1 or many instances.

But the point of designing using "Tell, Don't Ask" is that you get flexibility for free without having to think too much about it.